CRM/Contribute/Form/ContributionBase.php

```diff
@@ -413,10 +413,8 @@ public function preProcess() {
       // check for is_monetary status
       $isPayLater = $this->_values['is_pay_later'] ?? NULL;
       if ($this->getExistingContributionID()) {
-        $this->_values['financial_type_id'] = CRM_Core_DAO::getFieldValue('CRM_Contribute_DAO_Contribution',
-          $this->_ccid,
-          'financial_type_id'
-        );
+        $this->_values['financial_type_id'] = CRM_Core_DAO::getFieldValue('CRM_Contribute_DAO_Contribution', $this->_ccid, 'financial_type_id');
+        $this->_values['tax_amount'] = CRM_Core_DAO::getFieldValue('CRM_Contribute_DAO_Contribution', $this->_ccid, 'tax_amount');
```

Comment:
we wouldn't need to do this if we use `getContributionValue()` in the assign - because it is already available there & generally trying to move towards using that rather than the `_values` array - which is less predictable

Reply:
Thanks @eileenmcnaughton I don't knw about this magic fn `getContributionValue` . I have updated my PR